### PR TITLE
Add multiple exclusion criteria and notification of exclusion violations

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description
Adds multiple exclusion criteria and notification of exclusion violations

### Changelog

```
<changelog>Added multiple exclusion criteria and notification of exclusion violations.</changelog>
```

cc @zugaldia @truburt

### CODE REVIEW

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteExclusions.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * @see [RouteOptions.Builder.exclude]
+ */
+fun DirectionsRoute.exclusionViolations(): List<ExclusionViolation> {
+    val exclusionViolations = mutableListOf<ExclusionViolation>()
+    val excludeCriteria = ParseUtils.parseToStrings(this.routeOptions()?.exclude(), COMMA_DELIMETER)
+    excludeCriteria?.let {
+        this.legs()?.forEachIndexed { legIndex, leg ->
+            leg.steps()?.forEachIndexed { stepIndex, step ->
+                step.intersections()?.forEachIndexed { intersectionIndex, intersection ->
+                    intersection.classes()?.forEachIndexed { _, classes ->
```
Comment (korshaknn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteExclusions.kt:58]:
```suggestion
                    intersection.classes()?.forEach { classes ->
```

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteExclusions.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                                intersection
+                            )
+                            exclusionViolations.add(exclusionViolation)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return exclusionViolations.toList()
```
Comment (korshaknn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteExclusions.kt:77]:
```suggestion
    return exclusionViolations
```

**Review by korshaknn — APPROVED** [review]

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/route/RouteExclusionsJavaTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  @Test
+  public void tollAndFerryExclusionViolationsType() {
+    DirectionsRoute directionsRoute = DirectionsRoute.fromJson(
+        FileUtils.INSTANCE.loadJsonFixture("toll_and_ferry_directions_route.json")
+    );
+
+    Map<String, List<ExclusionViolation>> tollAndFerryExclusionViolations =
+        RouteExclusions.exclusionViolations(directionsRoute)
+            .stream()
+            .collect(groupingBy(ExclusionViolation::getType));
```
Comment (LukasPaczos) [inline libnavigation-base/src/test/java/com/mapbox/navigation/base/route/RouteExclusionsJavaTest.java:89]: Should we also expose this as an extension? :)

Can be a follow-up to consider.

**Review by LukasPaczos — APPROVED** [review]
:ship:

**Pull request merged**